src/atoms/Icon.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react'
+import {
+    findIconDefinition,
+    IconDefinition,
+    IconLookup,
+    IconName,
+    IconPrefix,
+    library
+} from '@fortawesome/fontawesome-svg-core'
+import { fab } from '@fortawesome/free-brands-svg-icons'
+import { far } from '@fortawesome/free-regular-svg-icons'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Styles } from '../constants/enum'
+
+library.add(fab, far, fas)
+
+interface IconProps {

```

Comment:
We can add more options like :
- The size : https://fontawesome.com/how-to-use/on-the-web/styling/sizing-icons
- The rotation : https://fontawesome.com/how-to-use/on-the-web/styling/rotating-icons
- The animation : https://fontawesome.com/how-to-use/on-the-web/styling/animating-icons